packages/scene-composer/src/components/three-fiber/AnimationComponent/__tests__/AnimationComponent.spec.tsx

```diff
@@ -0,0 +1,209 @@
+import * as THREE from 'three';
+import React, { useEffect } from 'react';
+import { render } from '@testing-library/react';
+
+import AnimationComponent, { toggleIsAnimationPaused, getAnimationObjectByName } from '../AnimationComponent';
+import { ISceneNodeInternal, useStore, IAnimationComponentInternal } from '../../../../store';
+import { generateUUID } from '../../../../utils/mathUtils';
+
+jest.mock('react', () => ({
+  ...jest.requireActual('react'),
+  useEffect: jest.fn(),
+}));
```

Comment:
try not to mock react hooks, this kind of tests are hard to maintain when they rely on matching certain call to certain value